components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/user/core/common/AbstractUserStoreManager.java

```diff
@@ -5924,9 +5924,13 @@ protected void addInitialAdminData(boolean addAdmin, boolean initialSetup) throw
                     log.error(message);
                 }
             } else if (addAdmin) {
+                String[] roleList = null;
                 try {
-                    this.doAddUser(adminUserName, realmConfig.getAdminPassword(),
-                            null, null, null, false);
+                    if (roleExist) {
+                        roleList = new String[]{getAdminRoleName()};
+                    }
+                    this.doAddUser(adminUserName, realmConfig.getAdminPassword(), roleList,
```

Comment:
We are adding the roles later in the flow[1]. Shall we check that?

[1] - https://github.com/wso2/product-micro-integrator/blob/master/components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/user/core/common/AbstractUserStoreManager.java#L6034

Reply:
https://github.com/wso2/product-micro-integrator/blob/master/components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/user/core/common/AbstractUserStoreManager.java#L3983 returns the true. So the Admin role updating part is skipped.

Reply:
Revert these changes as ICP does not support multiple super admins.